### PR TITLE
C api aliases

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will add `addAlias` methods to the C and C++98 API

### CODE REVIEW

File: src/helics/shared_api_library/helicsCore.h [hunk 1]
```diff
@@ -1380,6 +1380,17 @@ HELICS_EXPORT int helicsFederateGetIntegerProperty(HelicsFederate fed, int intPr
  */
 HELICS_EXPORT HelicsTime helicsFederateGetCurrentTime(HelicsFederate fed, HelicsError* err);
 
+/**
+ * create an alias for an interface
+ *
+ * @param fed The federate to use to set the alias
+ * @param interfaceName The current name of an interface
+ * @param alias the additional name to use for the given interface
```
Comment (nightlark) [inline src/helics/shared_api_library/helicsCore.h:1388]:
```suggestion
 * Create an alias for an interface.
 *
 * @param fed The federate to use to set the alias.
 * @param interfaceName The current name of an interface.
 * @param alias The additional name to use for the given interface.
```

File: src/helics/shared_api_library/helicsCore.h [hunk 2]
```diff
@@ -1554,6 +1565,28 @@ HELICS_EXPORT void helicsCoreSetGlobal(HelicsCore core, const char* valueName, c
  */
 HELICS_EXPORT void helicsBrokerSetGlobal(HelicsBroker broker, const char* valueName, const char* value, HelicsError* err);
 
+/**
+ * create an alias for an interface
+ *
+ * @param core The core to use to set the alias
+ * @param interfaceName The current name of an interface
+ * @param alias the additional name to use for the given interface
```
Comment (nightlark) [inline src/helics/shared_api_library/helicsCore.h:1573]:
```suggestion
 * Create an alias for an interface.
 *
 * @param core The core to use to set the alias.
 * @param interfaceName The current name of an interface.
 * @param alias The additional name to use for the given interface.
```

File: src/helics/shared_api_library/helicsCore.h [hunk 3]
```diff
@@ -1554,6 +1565,28 @@ HELICS_EXPORT void helicsCoreSetGlobal(HelicsCore core, const char* valueName, c
...
+ * @param[in,out] err A pointer to an error object for catching errors.
+ */
+HELICS_EXPORT void helicsCoreAddAlias(HelicsCore core, const char* interfaceName, const char* alias, HelicsError* err);
+
+/**
+ * create an alias for an interface
+ *
+ * @param broker The broker to use to set the alias
+ * @param interfaceName The current name of an interface
+ * @param alias the additional name to use for the given interface
```
Comment (nightlark) [inline src/helics/shared_api_library/helicsCore.h:1584]:
```suggestion
 * Create an alias for an interface.
 *
 * @param broker The broker to use to set the alias.
 * @param interfaceName The current name of an interface.
 * @param alias The additional name to use for the given interface.
```

File: src/helics/cpp98/Core.hpp [hunk 4]
```diff
@@ -131,6 +131,15 @@ class Core {
         helicsCoreSetGlobal(core, valueName.c_str(), value.c_str(), hThrowOnError());
     }
 
+    /** add a global alias for an interface
+    @param interfaceName the given name of the interface
+    @param alias the new name by which the interface can be referenced
+    */
+    void addAlias(const std::string& interfaceName, const std::string& alias)
+    {
+        helicsFederateAddAlias(core, interfaceName.c_str(), alias.c_str(), hThrowOnError());
```
Comment (nightlark) [inline src/helics/cpp98/Core.hpp:140]:
```suggestion
        helicsCoreAddAlias(core, interfaceName.c_str(), alias.c_str(), hThrowOnError());
```

File: src/helics/cpp98/Broker.hpp [hunk 5]
```diff
@@ -126,6 +126,14 @@ class Broker {
     {
         helicsBrokerSetGlobal(broker, valueName.c_str(), value.c_str(), hThrowOnError());
     }
+    /** add a global alias for an interface
+    @param interfaceName the given name of the interface
+    @param alias the new name by which the interface can be referenced
+    */
+    void addAlias(const std::string& interfaceName, const std::string& alias)
+    {
+        helicsFederateAddAlias(broker, interfaceName.c_str(), alias.c_str(), hThrowOnError());
```
Comment (nightlark) [inline src/helics/cpp98/Broker.hpp:135]:
```suggestion
        helicsBrokerAddAlias(broker, interfaceName.c_str(), alias.c_str(), hThrowOnError());
```

File: src/helics/cpp98/Core.hpp (hunk 4 above)
Comment (phlptp) [reply src/helics/cpp98/Core.hpp:140]: good catch, I suppose eventually we should have checks for every API call in the C++98 API as well as all the others

**Review by nightlark — APPROVED** [review]

**Pull request merged**